lib/libc/gen/dlfcn.c

```diff
@@ -192,17 +192,19 @@ _rtld_thread_init(void *li __unused)
 	/* Do nothing when linked statically. */
 }
 
+#ifdef __CHERI_PURE_CAPABILITY__
```

Comment:
I think I tried that and something failed to link, but my build tree may not have been sufficiently clean.

No ifdef is required in Symbol.map because it's not an error for a missing symbol to be listed (this is sometimes annoying). That being said, having a lib/libc/gen/Symbol-c18n.map or similar and adding it to SYM_MAPS conditionally in the Makefile could make sense.